### PR TITLE
Transition node to typescript in zenith

### PR DESCRIPTION
**Pull request by TomChv**
Replace #6373 with a simpler set of changes that only concerns our codebase and Zenith documentation.

#### List of updates:
- Our GitHub Actions (/cc @gerhard) 
- Our release doc (/cc @jedevc) 
- `sdk/nodejs` -> `sdk/typescript`
- Mage `sdk:nodejs` -> `sdk:typescript`
- Our generator
- I created a label `sdk/typescript` on GH

I also deleted the reference directory in `current_docs`, since these are generated files.

This PR should be way easier to review than the previous one since most changes are path related.

/cc @vikram-dagger @helderco 

Related to DEV-3212

### CODE REVIEW

**Comment by vikram-dagger** [comment]
Thanks for clearing out the auto-gen files. Speaking only for the docs, the changes seem fine. Will leave it to others to comment on the codebase changes.

File: sdk/typescript/README.md
Comment (jedevc) [inline sdk/typescript/README.md:7]:
```suggestion
The Dagger TypeScript SDK contains everything you need to develop CI/CD pipelines in TypeScript or Javascript, and run them on any OCI-compatible container runtime.
```

Comment (TomChv) [reply sdk/typescript/README.md]: Done in https://github.com/dagger/dagger/pull/6381/commits/5a635fc29b6b40c36fbe42fd315a12b29dd0a4be

File: RELEASING.md
Comment (TomChv) [reply RELEASING.md]: Changes in https://github.com/dagger/dagger/pull/6381/commits/c9bd9727c72287d557768a42984aaae081fd9218

**Review by jedevc — APPROVED** [review]

**Comment by jedevc** [comment]
LGTM once tests pass :heart:

**Pull request merged**
